### PR TITLE
Add KRUSTY 15 Cents Reactivity Insertion Model

### LINKED ISSUE #477: Add 15 Cent Reactivity Insertion to the KRUSTY Model
## Reason
The current KRUSTY model only covers steady state simulation using a coupled Griffin-BISON model. While a series of transient experiments were performed on KRUSTY, adding corresponding transient models to simulate these experiments is important for code/model validation.

## Design
The 15 cent Reactivity Insertion warm critical test model for KRUSTY will be added with documentation.

## Impact
The KRUSTY Multiphysics Model stands as an exemplary demonstration of high-fidelity multiphysics modeling for advanced microreactor designs intended for space exploration missions. The KRUSTY experiments provide precious data for multiphysic model V&V.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #477

### CODE REVIEW

**Comment by miaoyinb** [comment]
@GiudGiud This is to add a transient model to the existing KRUSTY model. Thanks in advance for your review.

File: doc/content/microreactors/KRUSTY/Griffin-BISON_Multiphysics_15C_Reactivity_Insertion_Test.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+## Griffin-BISON Multiphysics Model
+
+The multiphysics model to simulate the 15 Ȼ run is very similar to the one in the [previous VTB model for steady state calculations](/Griffin-BISON_Multiphysics_Steady_State_Model.md). Specifically, it employs a two-level MOOSE MultiApps hierarchy that tightly couples the Griffin neutronics model with the BISON thermos-mechanical model to simulate the transient. The DFEM-SN(2,3) solver with CMFD acceleration in Griffin was used to solve for fission power, and anisotropic scattering was modeled with NA=3. The calculated nuclear heat was transferred to BISON, where it was used to solve for core temperatures and displacement fields resulting from thermal expansions. These temperature and displacement fields were then transferred back to Griffin to update its cross sections and, consequently, the fission power. The coupling of Griffin and BISON was achieved through Picard iterations. To save on computational time and resources, only ¼ of the KRUSTY core was modeled.
```
Comment (GiudGiud) [inline doc/content/microreactors/KRUSTY/Griffin-BISON_Multiphysics_15C_Reactivity_Insertion_Test.md:13]:
```suggestion
The multiphysics model to simulate the 15 Ȼ run is very similar to the one in the [previous VTB model for steady state calculations](/Griffin-BISON_Multiphysics_Steady_State_Model.md). Specifically, it employs a two-level MOOSE MultiApps hierarchy that tightly couples the Griffin neutronics model with the BISON thermo-mechanical model to simulate the transient. The DFEM-SN(2,3) solver with CMFD acceleration in Griffin was used to solve for fission power, and anisotropic scattering was modeled with NA=3. The calculated nuclear heat was transferred to BISON, where it was used to solve for core temperatures and displacement fields resulting from thermal expansions. These temperature and displacement fields were then transferred back to Griffin to update its cross sections and, consequently, the fission power. The coupling of Griffin and BISON was achieved through Picard iterations. To save on computational time and resources, only ¼ of the KRUSTY core was modeled.
```

File: doc/content/microreactors/KRUSTY/Griffin-BISON_Multiphysics_15C_Reactivity_Insertion_Test.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+!media media/KRUSTY/reactivity_insertion.png
+      style=display: block;margin-left:auto;margin-right:auto;width:60%;
+      id=reactivity_insert
+      caption= Displacement field caused by lifting the radial reflector to insert the 15 Ȼ reactivity
+
+The upward movement of the radial reflector was modeled in BISON by applying a Dirichlet boundary condition to the bottom surface of the solid assembly. BISON calculated the displacement field by solving tensor mechanics equations. The BISON inputs used to generate the displacement are included   in Appendix A. Figure 1 presents the calculated displacement field which is identical to Figure 2 in [previous VTB results](/Neutronic_Multiphysics_Steady_State_Results.md). To make more efficient use of computer resources (by enabling more cores to be utilized per node), the KRTUSTY mesh has been pre-split into chunks and was distributed for parallel computation.
```
Comment (GiudGiud) [inline doc/content/microreactors/KRUSTY/Griffin-BISON_Multiphysics_15C_Reactivity_Insertion_Test.md:22]:
```suggestion
The upward movement of the radial reflector was modeled in BISON by applying a Dirichlet boundary condition to the bottom surface of the solid assembly. BISON calculated the displacement field by solving the solid mechanics equations. Figure 1 presents the calculated displacement field which is identical to Figure 2 in [previous VTB results](/Neutronic_Multiphysics_Steady_State_Results.md). To make more efficient use of computer resources (by enabling more cores to be utilized per node), the KRUSTY mesh has been pre-split into chunks and was distributed for parallel computation.
```

**Review by GiudGiud — COMMENTED** [review]
you ll want to add a few tags to the main krusty page, mainly RIA ?

any ideas for testing? can we regenerate the checkpoint file in a testing run (ideal) ?

**Comment by miaoyinb** [comment]
> you ll want to add a few tags to the main krusty page, mainly RIA ?
> 
> any ideas for testing? can we regenerate the checkpoint file in a testing run (ideal) ?

RIA might not be the ideal term as this is a warm critical test instead of an accident test. Maybe just use "warm critical test"?

Testing could be tricky as it will rely on the steady-state simulation checkpoint/SolutionVector output files, which also depend on the mpi ranks used. Any suggestion?

**Comment by GiudGiud** [comment]
Running the steady case then running the transient is the easiest. 

>  which also depend on the mpi ranks used. Any suggestion?

alternatively, if we LFS checked in the steady checkpoint we would have to be mindful of that

**Comment by GiudGiud** [comment]
there s a model under "Reactivity Insertion" already, for MR drum. could go with that one?

**Comment by miaoyinb** [comment]
> Running the steady case then running the transient is the easiest.

It takes ~4 hours on 10 sawtooth nodes to finish SN(2,3) NA=3 steady state.

Maybe we can try SN(1,3) NA=1 and force it to finish after one Richardson iteration and start the transient to run one timestep?

**Comment by GiudGiud** [comment]
we can try something like that though we wont see a steady state in the beginning of the transient

**Comment by miaoyinb** [comment]
Let's try to see how long it takes to run such a case.

**Comment by miaoyinb** [comment]
Good news: SN(1,3) NA=1 steady state with one Richardson iteration takes ~1000 seconds to run using 40 cores on a sawtooth node. 
Bad news: the corresponding transient case using SN(1,3) NA=1 does not have enough memory to run on a single node. (it needs more nodes + distributed mesh to relieve the memory burden on each node.)

Maybe we can run the "simplified" steady-state case and then do syntax check only for the transient?

**Comment by GiudGiud** [comment]
can we reduce the mesh maybe?

**Comment by miaoyinb** [comment]
Let me try

**Comment by miaoyinb** [comment]
Reducing axial mesh density works. Using SN(1,3) NA=1 and allowing Richardson iteration to converge, the steady-state run takes ~2 hours and one time step transient takes about 500 seconds. I guess, this is okay for a "heavy" test.

**Comment by miaoyinb** [comment]
Added the tests. It seems the syntax override+!include does not like `cli_args`. I had to add an input file for testing.

**Comment by miaoyinb** [comment]
@GiudGiud Let me know if these tests are okay or if you have any additional comments. Thanks.

**Comment by GiudGiud** [comment]
Will do.
Need to get a submodule update done first

File: microreactors/KRUSTY/tests [hunk 3]
```diff
@@ -83,4 +83,52 @@
...
+      cli_args = "--mesh-only MESH/BISON_mesh_coarse.e
+                 Mesh/gen/file='Griffin_mesh_coarse.e'"
+      prereq = 'transient_tests/mesh_gen_griffin_coarse'
+    []
+    [multiapp_ss_coarse]
+      type = CSVDiff
+      input = 'Multiphysics_SS/KRUSTY_Griffin_SN23_NA23_CMFD_coarse.i'
+      csvdiff = 'Multiphysics_SS/KRUSTY_Griffin_SN23_NA23_CMFD_coarse_out.csv'
+      executable_pattern = 'blue_crab*|dire_wolf*'
+      min_parallel = 40
```
Comment (GiudGiud) [inline microreactors/KRUSTY/tests:108]: this is still too much for us (max 16)

I ll catch this under HPC testing. It is reasonable there. Thanks for setting it up

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**